recipes/paho-mqtt-c/all/patches/0002-fix-MinGW-and-OSX-builds-for-1-3-5.patch

```diff
@@ -0,0 +1,49 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index b53745b..d8787d8 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -41,7 +41,7 @@
+ ENDIF()
+ 
+ IF(WIN32)
+-  ADD_DEFINITIONS(-D_CRT_SECURE_NO_DEPRECATE -DWIN32_LEAN_AND_MEAN -MD)
++  ADD_DEFINITIONS(-D_CRT_SECURE_NO_DEPRECATE -DWIN32_LEAN_AND_MEAN)
+ ELSEIF(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+   ADD_DEFINITIONS(-DOSX)
+ ENDIF()
+@@ -73,33 +73,6 @@
+     ADD_SUBDIRECTORY(doc)
+ ENDIF()
+ 
+-IF (PAHO_ENABLE_CPACK)
```

Comment:
do we really need to remove this? maybe just add `cmake.defintions["PAHO_ENABLE_CPACK"] = False`?

Reply:
it was apparently removed to avoid the message:

    [HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown folder 'share' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013) 

I haven't yet found another way of avoiding it.  I can look further for the next release.

Reply:
Don't change your project because of it, you can simply remove the share folder after installing. Patches are worst because they can become incompatible in the future.

```python
def package(self):
    # after installing all stuff
    tools.rmdir(os.path.join(self.package_folder, "share"))